src/main/java/dk/bec/polonez/reservationsystem/controller/DiscountController.java

```diff
@@ -14,8 +18,33 @@ public DiscountController(DiscountService discountService) {
         this.discountService = discountService;
     }
 
-    @GetMapping("dc/{code}")
-    public int getByCode(String code){
+    @GetMapping("{code}")
+    public DiscountDto getByCode(@PathVariable String code) {
         return discountService.getByCode(code);
     }
+
+    @GetMapping("{id}")
+    public DiscountDto getById(@PathVariable long id) {
+        return discountService.getById(id);
+    }
+
+    @GetMapping
+    public List<DiscountDto> getAll() {
+        return discountService.getAll();
+    }
+
+    @DeleteMapping("{id}")
+    public DiscountDto deleteDiscount(@PathVariable long id) {
+        return discountService.deleteDiscount(id);
+    }
+
+    @PutMapping("{id}")
+    public DiscountDto updateDiscount(@PathVariable long id, @RequestBody CreateDiscountDto discountDto) {
+        return discountService.updateDiscount(id, discountDto);
+    }
+
+    @PostMapping("{id}")
```

Comment:
**advice** ```{id}``` in POST is suspicious, shouldn't it be removed as redundant (not used)?